src/apps/chahub/tasks.py

```diff
@@ -1,56 +1,165 @@
+    except requests.exceptions.RequestException:
+        resp = None
+
+    if resp and resp.status_code in (200, 201):
+        logger.info(f"ChaHub :: Received response {resp.status_code} {resp.content}")
+        obj.chahub_timestamp = timezone.now()
+        obj.chahub_data_hash = data_hash
+        obj.chahub_needs_retry = False
+    else:
+        status = getattr(resp, 'status_code', 'N/A')
+        body = getattr(resp, 'content', 'N/A')
+        logger.info(f"ChaHub :: Error sending to chahub, status={status}, body={body}")
+        obj.chahub_needs_retry = True
+    obj.save(send=False)
+
+
+@app.task(queue='site-worker')
+def delete_from_chahub(app_label, pk):
+    if not settings.CHAHUB_API_URL:
+        raise ChahubException("CHAHUB_API_URL env var required to send to Chahub")
+    if not settings.CHAHUB_API_KEY:
+        raise ChahubException("No ChaHub API Key provided")
+
+    obj = get_obj(app_label, pk, include_deleted=True)
+
+    url = f"{settings.CHAHUB_API_URL}{obj.get_chahub_endpoint()}{pk}/"
+    logger.info(f"ChaHub :: Sending to ChaHub ({url}) delete message")
+
+    headers = {'X-CHAHUB-API-KEY': settings.CHAHUB_API_KEY}
+
+    try:
+        resp = requests.delete(url=url, headers=headers)
+    except requests.exceptions.RequestException:
+        resp = None
+
+    if resp and resp.status_code == 204:
+        logger.info(f"ChaHub :: Received response {resp.status_code} {resp.content}")
+        obj.delete()
+    else:
+        status = getattr(resp, 'status_code', 'N/A')
+        body = getattr(resp, 'content', 'N/A')
+        logger.info(f"ChaHub :: Error sending to chahub, status={status}, body={body}")
+        obj.chahub_needs_retry = True
+        obj.save(send=False)
+
+
+def batch_send_to_chahub(model, limit=None, retry_only=False):
+    qs = model.objects.all()
+    if retry_only:
+        qs = qs.filter(chahub_needs_retry=True)
+    if limit is not None:
+        qs = qs[:limit]
+
+    endpoint = model.get_chahub_endpoint()
+    data = [obj.get_chahub_data() for obj in qs if obj.get_chahub_is_valid()]
+    if not data:
+        logger.info(f'Nothing to send to Chahub at {endpoint}')
+        return
     try:
-        logger.info("Sending profile data to Chahub")
-        resp = send_to_chahub('profiles/', user_data_list, update=False)
+        logger.info(f"Sending all data to Chahub at {endpoint}")
+        resp = _send(endpoint=endpoint, data=data)
         logger.info(f"Response Status Code: {resp.status_code}")
-    # TODO: Will this catch timeouts and errors from our code? We should bubble up errors from send_to_chahub nicely.
+        if resp.status_code != 201:
+            logger.warning(f'ChaHub Response Content: {resp.content}')
     except ChahubException:
-        logger.info("There was a problem reaching Chahub, it is currently offline. Re-trying in 5 minutes.")
-        send_users_to_chahub.apply_async(eta=timezone.now() + timedelta(minutes=5))
```

Comment:
should we retry this in 5 minutes? do we have a task to retry these already somewhere?

Reply:
yep we do, `do_chahub_retries()`